cobigen/cobigen-tsplugin/src/main/java/com/devonfw/cobigen/tsplugin/merger/TypeScriptMerger.java

```diff
@@ -63,107 +86,125 @@ public String getType() {
+            throw new RuntimeException("Failed : HTTP error code : " + conn.getResponseCode());
         }
     }
 
     /**
      * Reads the output.ts temporary file to get the merged contents
-     * @param base
-     *            base file just for exception handling
-     * @param mergedContents
-     *            merged typescript code
+     * @param importsAndExports
+     *            The part of the code where imports and exports are declared
+     * @param body
+     *            the rest of the body of the source file
      * @return merged contents already beautified
      */
-    private String runBeautifierExcludingImports(File base, String mergedContents) {
-        StringBuilder importsAndExports = new StringBuilder();
-        StringBuilder body = new StringBuilder();
+    private String runBeautifierExcludingImports(String importsAndExports, String body) {
 
-        try (StringReader inR = new StringReader(mergedContents); BufferedReader br = new BufferedReader(inR)) {
-            String line;
-            while ((line = br.readLine()) != null) {
-                if (line.startsWith("import ") || isExportStatement(line)) {
-                    importsAndExports.append(line);
-                    importsAndExports.append(LINE_SEP);
-                } else {
-                    body.append(line);
-                    body.append(LINE_SEP);
-                }
-            }
-        } catch (IOException e) {
-            throw new MergeException(base, "Could not process merged contents for formatting.", e);
-        }
+        FileTO fileTO = new FileTO(body);
+        HttpURLConnection conn = request.getConnection("POST", "Content-Type", "application/json", "beautify");
+
+        StringBuffer bodyBuffer = new StringBuffer();
+        try (OutputStream os = conn.getOutputStream(); OutputStreamWriter osw = new OutputStreamWriter(os, "UTF-8");) {
+
+            sendRequest(fileTO, conn, os, osw);
 
-        String formattedBody =
-            executeJS(base, invocable -> invocable.invokeMethod(((ScriptEngine) invocable).eval("global"),
-                "js_beautify", body.toString()), Constants.BEAUTIFY_JS);
+            BufferedReader br = new BufferedReader(new InputStreamReader((conn.getInputStream())));
+
+            LOG.info("Receiving output from Server....");
+            Stream<String> s = br.lines();
+            s.parallel().forEachOrdered((String line) -> {
+                bodyBuffer.append(line);
+                bodyBuffer.append(LINE_SEP);
+            });
+
+        } catch (IllegalStateException e) {
+
+            LOG.error("Closing connection on InputReader.", e);
+            request.terminateProcessConnection();
+        } catch (Exception e) {
+            connectionExc.setConnectExceptionMessage("Connection to server failed, attempt number " + 0 + ".");
+            connectionExc.setIOExceptionMessage("IO exception when merging");
+
+            connectionExc.handle(e);
+        }
```

Comment:
I think there is a great potential to share the http connection code in a separate method serving just url to be called and the parameters to be passed serving back the result.

Reply:
I think that I have implemented something similar with:

`HttpURLConnection conn = request.getConnection("POST", "Content-Type", "application/json", "beautify");`

`getConnection(...)` is a method I implemented [here](https://github.com/devonfw/tools-cobigen/pull/840/files#diff-3c4d8d9150d5465bd1104dbbf44849b4R271).